src/entity.js

```diff
@@ -0,0 +1,99 @@
+function runTransforms(transforms, entry) {
+  const [originalKey, originalValue] = entry
+
+  const transform = transforms.find(({ test }) =>
+    new RegExp(test).test(originalKey)
+  )?.transform
+
+  if (transform == null) return [originalKey, originalValue]
+
+  const result = transform(originalValue, originalKey)
+  if (Array.isArray(result) && result.length <= 2) {
+    const [transfomedValue, transfomedKey] = result
+    return [transfomedKey ?? originalKey, transfomedValue]
+  }
+  return [originalKey, result]
+}
+
+const createDescriptorFor = (
+  key,
+  {
+    initialValues,
+    modifiedValues,
+
+    enumerable = true,
+    configurable = true,
+    writable = true,
+  } = {}
+) => {
+  const descriptor = {
+    enumerable,
+    configurable,
+  }
+
+  if (writable) {
+    descriptor.get = () =>
+      (Object.prototype.hasOwnProperty.call(modifiedValues, key)
+        ? modifiedValues
+        : initialValues)[key]
+
+    descriptor.set = newValue => {
+      modifiedValues[key] = newValue
+    }
+  } else descriptor.get = () => initialValues[key]
+
+  return descriptor
+}
+
+class Entity {
+  #options
+
+  #initialValues = {}
+
+  #modifiedValues = {}
+
+  constructor(initialState, options) {
+    this.#options = {
+      ...this.constructor.defaultOptions,
+      ...options,
+    }
```

Comment:
IMHO you should be extending `transforms` instead of overriding it.

Something like this but more sophisticated and bullet proof:

```jsx
  constructor(initialState, options) {
    this.#options = {
      ...this.constructor.defaultOptions,
      ...options,	
	  transforms: [
      	...options.transforms, // based on this commend https://github.com/oacore/api-client/pull/43/files#r378142558
      	...this.constructor.defaultOptions.transforms,
		]
    }
```

Reply:
There may be transforms you want to override.

If I override, it's still possible overrider with `[...myTransforms, ...Entity.defaultOptions.tranforms]` but if I extend there is no way to override.